Tests/SwiftAlgorithmsTests/PartitionTests.swift

```diff
@@ -88,4 +88,42 @@ final class PartitionTests: XCTestCase {
       }
     }
   }
+  
+  func testPartitionWithSubrangeBidirectionalCollection() {
+    for length in 10...20 {
+      let a = Array(0..<length)
+      for j in 0..<length {
+        var b = a
+        let partitionRange = 0..<j
```

Comment:
Looks like this is only testing the subrange at the beginning of the collection — could we iterate over `0...j` as well, for the start of the subrange?

Reply:
Right, I've updated!
Let me know if that is what you have in mind  :)

Reply:
IIUC I'm looking for one more nested `for` loop inside this one, a `for k in 0...j`, and then `let partitionRange = k..<j` (with hopefully some better variable names).

The issue is that, when `a` has 10 elements, we're partitioning all the subranges starting at `0`, and also now the subranges ending at `length`, below. What we aren't getting is the middle subranges, like `2..<6`.

Reply:
Ah I see, makes sense. 
I'm on it, thank you!

Reply:
Done!
Also, adjusted for stable partition tests :)

Reply:
Ping @natecook1000 =]